pykechain/client.py

```diff
@@ -1556,6 +1556,43 @@ def _create_parts_bulk(
         parts = self.parts(id__in=",".join(part_ids))
         return PartSet(parts=parts)
 
+    def _delete_parts_bulk(
+            self,
+            parts: List[Union[Part2]],
+            asynchronous: Optional[bool] = False,
+            **kwargs
+    ) -> bool:
+        """Delete multiple Parts simultaneously.
+
+        :param parts: list of Part objects or UUIDs
+        :type parts: List[Property] or List[UUID]
+        :param asynchronous: If true, immediately returns (default = False)
+        :type asynchronous: bool
+        :param kwargs:
+        :return: True if parts are delete successfully
+        :raises APIError: if the parts could not be deleted
+        :raises IllegalArgumentError: if there were neither Parts nor UUIDs in the list of parts
+        """
+        list_parts = list()
+        for part in parts:
+            if isinstance(part, Part2):
```

Comment:
Here is a mix of processing the data passed, and validating it at the same time. Don't you want to separate those?

Reply:
I thought it would be more efficient to do it together, as I am already looping over the parts in the first place.